src/lxc/monitor.c

```diff
@@ -181,7 +181,6 @@ int lxc_monitor_sock_name(const char *lxcpath, struct sockaddr_un *addr) {
 	if (ret < 0)
 		return -1;
 
-	sockname[sizeof(addr->sun_path)-3] = '\0';
```

Comment:
I think that only the last one should be set to `\0`. so we change it to  `sockname[sizeof(addr->sun_path)-2] = '\0';`  or `addr->sun_path[sizeof(addr->sun_path)-1] = '\0';`